### PR TITLE
TLS/ACL Support

### PR DESCRIPTION
**Pull request by tremblay-stripe**
Smokescreen is going to receive HTTP CONNECT proxy requests from Envoy. It has to understand TLS.

### CODE REVIEW

**Comment by rlk-stripe** [comment]
Also, In general, I'm not a huge fan of the name "smoker", as it's kind of meaningless.  Also, I think that assigning an explicit name for an import (when not specifically required by e.g. a naming conflict) is probably not idiomatic for Go.

File: cmd/smokescreen.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	if args == nil {
+		err = app.Run(os.Args)
+	} else {
+		err = app.Run(args)
+	}
+
+	return configToReturn, err
+}
+
+func addCidrToSlice(blocks []net.IPNet, cidrBlockString string) ([]net.IPNet, error) {
```
[the remark this answers is not in the document]
Comment (tremblay-stripe) [reply cmd/smokescreen.go]: You have a very sharp eye. I knew I had to find the proper place for this to live but totally forgot about it. Thanks!

**Comment by tremblay-stripe** [comment]
> https://github.com/stripe/smokescreen/pull/14#issuecomment-415833717
I renamed the package. I took that from slashask's `asker`. I do agree with you.

File: cmd/testdata/pki/ca.pem [hunk 2]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN CERTIFICATE-----
```
[the remark this answers is not in the document]
Comment (tremblay-stripe) [reply cmd/testdata/pki/ca.pem]: The purpose behind it was to make sure it was obvious that is phony

File: cmd/smokescreen.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+		cli.StringSliceFlag{
+			Name:  "tls-client-ca",
+			Usage: "Root Certificate Authority used to authenticate clients",
+		},
+		cli.StringSliceFlag{
+			Name:  "crls",
+			Usage: "CRL used by the server (reloaded upon change)",
+		},
+		cli.BoolFlag{
+			Name:  "danger-allow-access-to-private-ranges",
```
[the remark this answers is not in the document]
Comment (rlk-stripe) [reply cmd/smokescreen.go]: Oops, the `cidr-blacklist` option (aka `deny-range` in my PR) *adds* items to the existing list, so you do have to exempt everything to duplicate this functionality.

**Pull request merged**